More fix for aot_export_module name collision during unlifting

yushangdi opened the pull request:
Summary: Also check the module's named buffers and parameters when resolving name collision

Test Plan:
```
buck2 run mode/dev-nosan caffe2/test/inductor:test_aot_inductor -- -r aoti_constant_tensor_name_collision
```

Differential Revision: D73264885




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

angelayi commented on torch/_inductor/compile_fx.py:
```diff
@@ -261,11 +261,21 @@ def _resolve_name_collision(mod: GraphModule, gm: GraphModule) -> None:
                 i = max(i, int(node.target.split(prefix)[-1]))
+        for key in existing_keys:
+            if key.startswith(prefix):
+                i = max(i, int(key.split(prefix)[-1]))
```
I just realized, what if the target is like "tensor_constant_foo"? wouldn't the `int(node.target.split(prefix)[-1])` fail?

yushangdi replied:
@angelayi  I don't think this will happen, because we only call `find_smallest_i` when there is a name collision, and name collision only happens when `aot_export_module` create attribute names.

`aot_export_module` only create attribute names that are like `tensor_constant_<int>`. 

I can change this to be more robust though.

yushangdi replied:
@angelayi Fixed to be more robust now. If the post_fix is not a digit, we'll just skip it when finding the next number for `i`.

yushangdi committed:
More fix for aot_export_module name collision during unlifting (#151684)

Summary:

Also check the module's named buffers and parameters when resolving name collision

Test Plan:
```
buck2 run mode/dev-nosan caffe2/test/inductor:test_aot_inductor -- -r aoti_constant_tensor_name_collision
```

Reviewed By: angelayi

Differential Revision: D73264885

yushangdi commented:
@pytorchbot merge -f "merged internally"
